### ISSUE TITLE
Numerical inconsistency between tensor(0+0j)**tensor(0) and tensor(0+0j)**0

### ISSUE THREAD

**Issue by nexuslrf** [pytorch/pytorch#91790]
### 🐛 Describe the bug

Although this scenario is less likely to happen, such a bug could ruin the model training if it happens.

```python
import torch
import numpy as np

print('torch:', torch.__version__)

print(np.array(0+0j)**0)
print(torch.tensor(0+0j)**0)
print(torch.tensor(0+0j)**torch.tensor(0))
```

Output
```
torch: 1.13.0+cu116
(1+0j)
tensor(1.+0.j)
tensor(nan+nanj)
```

### Versions

```
PyTorch version: 1.13.0+cu116
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.6
Libc version: glibc-2.27

Python version: 3.8.16 (default, Dec  7 2022, 01:12:13)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.10.147+-x86_64-with-glibc2.27
Is CUDA available: False
CUDA runtime version: 11.2.152
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.13.0+cu116
[pip3] torchaudio==0.13.0+cu116
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.14.0
[pip3] torchvision==0.14.0+cu116
[conda] Could not collect
```

cc @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

**Comment by lezcano** [comment]
Note that, mathematically, 0**0 is undefined, so I think that if one thinks that their model may hit this issue, they should properly mask the result to return whatever they think it's sensible given their model requirements.

**Comment by mingfeima** [comment]
@CaoE shall we add arg check?

**Comment by lezcano** [comment]
I think there's nothing actionable here as the operation is mathematically undefined here. Also, the result agrees in CPU and CUDA.
The user should not rely on these sort of results.
